wear/src/main/java/io/homeassistant/companion/android/home/views/HomeView.kt

```diff
@@ -224,21 +228,47 @@ fun LoadHomePage(
                     }
                 )
             }
-            composable(SCREEN_SET_TILE_TEMPLATE) {
+            composable("$ROUTE_TEMPLATE_TILE/$SCREEN_SELECT_TEMPLATE_TILE") {
+                SelectTemplateTileView(
+                    templateTiles = mainViewModel.templateTiles,
+                    onSelectTemplateTile = { tileId ->
+                        swipeDismissableNavController.navigate("$ROUTE_TEMPLATE_TILE/$tileId/$SCREEN_SET_TILE_TEMPLATE")
+                    }
+                )
+            }
+            composable(
+                route = "$ROUTE_TEMPLATE_TILE/{$ARG_SCREEN_TEMPLATE_TILE_ID}/$SCREEN_SET_TILE_TEMPLATE",
+                arguments = listOf(
+                    navArgument(name = ARG_SCREEN_TEMPLATE_TILE_ID) {
+                        type = NavType.StringType
+                    }
+                )
+            ) { backStackEntry ->
+                val tileId = backStackEntry.arguments!!.getString(ARG_SCREEN_TEMPLATE_TILE_ID)!!.toIntOrNull()
+                val templateTileConfig = mainViewModel.templateTiles[tileId]
+
                 TemplateTileSettingsView(
-                    templateContent = mainViewModel.templateTileContent.value,
-                    refreshInterval = mainViewModel.templateTileRefreshInterval.value
+                    templateContent = templateTileConfig?.template ?: "",
```

Comment:
I believe we may need to access the view model variables directly here in the function instead of the variable defined at line 248 which may lead to static data. Not sure if your last comment suggested you had tried this already?

Reply:
Ah, 2 months later I think I understand what you meant here. No, IIRC I did not try what I think you're suggesting here. I'll do that now. :drum:

Reply:
It didn't seem to help :unamused: Pushed the change now as 609ea01.

Reply:
did you confirm the variables had the expected data? as long as data is updating and the viewmodel variable is updating as things get added it should work just like the shortcut tile works

Reply:
I think I eventually figured it out late last night. `PhoneSettingsListener` and `TemplateTile` only ever update `WearPrefsRepository`, and changes in it are never propagated to `MainViewModel`. I haven't fixed it yet, but I think I understand the problem now at least.

Reply:
What would be the proper way to solve this? 🤔 I thought I followed prior art but maybe this is broken for shortcut tiles too.

Reply:
I tested the following now, but I honestly don't understand how it can work:

- Opened the HA wear app on my watch and went to the settings and list of shortcut tiles (one previously added tile listed)
- Without killing the HA wear app, I went to the home screen and added a new shortcut tile
- Went back to the HA wear app and opened the shortcut tile settings again
- The new shortcut tile is listed :exploding_head: 

AFAICT, `ShortcutsTile` does not call anything else than `WearPrefsRepository` from `onTileAddEvent`, and still the change is picked up by `MainViewModel`. How on earth is that possible? What am I missing here?

Reply:
> I tested the following now, but I honestly don't understand how it can work:
> 
> * Opened the HA wear app on my watch and went to the settings and list of shortcut tiles (one previously added tile listed)
> * Without killing the HA wear app, I went to the home screen and added a new shortcut tile
> * Went back to the HA wear app and opened the shortcut tile settings again
> * The new shortcut tile is listed 🤯

Thats exactly how it works with Camera tile as well. When the user clicks on Shortcut tile settings they are loaded again.

https://github.com/home-assistant/android/blob/2c20baf0f54897c720099b34e57cf78eaa70fedb/wear/src/main/java/io/homeassistant/companion/android/home/views/HomeView.kt#L143

Camera tile is a state flow so anytime the database is updated the other values update as well. With Camera tiles you can keep the camera tile setting page open and you'll see new camera tiles being added when you return back to the screen.

https://github.com/home-assistant/android/blob/2c20baf0f54897c720099b34e57cf78eaa70fedb/wear/src/main/java/io/homeassistant/companion/android/home/MainViewModel.kt#L94

Reply:
> When the user clicks on Shortcut tile settings they are loaded again.
> https://github.com/home-assistant/android/blob/2c20baf0f54897c720099b34e57cf78eaa70fedb/wear/src/main/java/io/homeassistant/companion/android/home/views/HomeView.kt#L143

Oh.. That one I missed :facepalm: Thanks, with 23e50710 it seems to finally work :tada: